cmd/config-migrate.go

```diff
@@ -27,6 +28,39 @@ import (
 // DO NOT EDIT following message template, please open a github issue to discuss instead.
 var configMigrateMSGTemplate = "Configuration file %s migrated from version '%s' to '%s' successfully.\n"
 
+var errConfigOld = errors.New("configuration version is older than expected")
+
+// Check if the config version is the latest,
+// returns nil if no migration needed, otherwise
+// proceed to migrate.
+func checkConfigForMigrate() error {
+	configFile := getConfigFile()
+
+	type serverConfigVersion struct {
+		Version string
+	}
+
+	srvCfgVersion := &serverConfigVersion{}
+	version, err := quick.GetVersion(configFile, srvCfgVersion)
```

Comment:
Any quick friendly configuration file must have `Version` field.  There is no need to define a struct type to do here.  Its quick package responsibility to read version from the file. I still suggest `quick.GetVersion(configFile string)`